lib/IDE/CompletionLookup.cpp

```diff
@@ -2440,8 +2440,13 @@ void CompletionLookup::getPostfixKeywordCompletions(Type ExprType,
       if (instanceTy->isAnyExistentialType()) {
         addKeyword("Protocol", MetatypeType::get(instanceTy),
                    SemanticContextKind::CurrentNominal);
-        addKeyword("Type", ExistentialMetatypeType::get(instanceTy),
-                   SemanticContextKind::CurrentNominal);
+        if (instanceTy->hasParenSugar()) {
+          addKeyword("Type", MetatypeType::get(instanceTy),
+                     SemanticContextKind::CurrentNominal);
+        } else {
+          addKeyword("Type", ExistentialMetatypeType::get(instanceTy),
+                     SemanticContextKind::CurrentNominal);
+        }
```

Comment:
Yeah, this is the right direction. But there really is no guarantee that the semantic type will retain the parentheses. A better option is to check the parsed type representation, which is meant to preserve the structure of the written type. You can get it from the type expression.

Reply:
Wouldn't this check ensure that?

```C++
if (isa<TypeExpr>(ParsedExpr)) {
  // ...
}
```

Reply:
I think you misunderstand. I am suggesting to perform the same checks on the type representation, not the semantic type. The type representation is stored in the type expression.

Reply:
I think the existing function is optimal here.